python/cudf/cudf/__init__.py

```diff
@@ -1,9 +1,58 @@
 # Copyright (c) 2018-2019, NVIDIA CORPORATION.
+""" __init__.py
 
+   isort:skip_file
+"""
 import cupy
 
-import rmm
+from cudf.errors import UnSupportedGPUError, UnSupportedCUDAError
+
+gpus_count = cupy.cuda.runtime.getDeviceCount()
+
+
+if gpus_count > 0:
+    for device in range(0, gpus_count):
+        # cudaDevAttrComputeCapabilityMajor - 75
+        major_version = cupy.cuda.runtime.deviceGetAttribute(75, device)
+
+        if major_version >= 6:
+            # You have a GPU with NVIDIA Pascal™ architecture or better
+            pass
+        else:
+            raise UnSupportedGPUError(
+                "You will need a GPU with NVIDIA Pascal™ architecture or better"
+            )
+
+    cuda_runtime_version = cupy.cuda.runtime.runtimeGetVersion()
 
+    if cuda_runtime_version > 10000:
+        # CUDA Runtime Version Check: Runtime version is greater than 10000
+        pass
+    else:
+        raise UnSupportedCUDAError(
+            "Please update your CUDA Runtime to 10.0 or above"
+        )
+
+    cuda_driver_version = cupy.cuda.runtime.driverGetVersion()
+
+    if cuda_driver_version == 0:
+        raise UnSupportedCUDAError("Please install CUDA Driver")
+    elif cuda_driver_version >= cuda_runtime_version:
+        # CUDA Driver Version Check: Driver Runtime version is >= Runtime version
+        pass
+    else:
+        raise UnSupportedCUDAError(
+            "Please update your CUDA Driver to 10.0 or above"
```

Comment:
https://docs.nvidia.com/deploy/cuda-compatibility/index.html#binary-compatibility
```suggestion
            "Please update your NVIDIA GPU Driver version to >=410.48"
```

Reply:
Actually, this isn't quite right. If they have CUDA 10.2, then they needed a newer driver >= 440.33.

So really, the error message should be customized based on `cuda_runtime_version` using the compatibility table I linked above.

CUDA 10.2 (10.2.89) | >= 440.33
-- | --
CUDA 10.1 (10.1.105) | >= 418.39
CUDA 10.0 (10.0.130) | >= 410.48

Reply:
This mapping is going to be problematic to maintain at a source code level. It would be nice if this was factored out and moved to some sort of configuration file that can be changed whenever we add or drop support for a CUDA version. The configuration could also list deprecated versions and then we would have the option of warning when a deprecated CUDA version is installed.

Reply:
@jrhemstad and I had a discussion offline on maintaining this hard-coded table yesterday and decided to throw an informative message instead like we do currently in this PR and point to the URL where these tables are updated when there is a change in cuda or driver versions. We thought of the second approach purely based on maintenance purpose but let me know if we would like to have such kind of table. Also as @kkraus14 pointed out in one of the comment reg architecture of GPUs https://github.com/rapidsai/cudf/pull/4692#discussion_r398885659 do we have any such thing in case of CUDA as well?